flytekit/common/launch_plan.py

```diff
@@ -116,12 +116,13 @@ def fetch(cls, project, domain, name, version=None):
 
         sdk_lp = cls.promote_from_model(lp.spec)
         sdk_lp._id = lp.id
+        sdk_lp.assign_name(name)
```

Comment:
is name not merely a shortcut to the name field in the ID? Or is there another action taken as part of assigning a name?

Reply:
another field is set, something that is read in sdk_dynamic.